abe/unittest.py

```diff
@@ -107,7 +107,32 @@ def assert_headers_contain(self, response_data, spec_data):
                 "header {0}".format(expected_header)
             )
 
-    def assert_matches_sample(self, path, label, url, response):
+    def assert_matches_request(self, sample_request, wsgi_request):
+        """
+        Check that the sample request and wsgi request match.
+        """
+        self.assertEqual(wsgi_request.META['PATH_INFO'], sample_request['url'])
+        self.assertEqual(wsgi_request.META['REQUEST_METHOD'],
+                         sample_request['method'])
+
+        if 'headers' in sample_request:
+            self.assert_headers_contain(
+                wsgi_request.META, sample_request['headers']
+            )
+
+        if 'body' in sample_request:
+            self.assert_data_equal(wsgi_request.POST, sample_request['body'])
+
+    def assert_matches_response(self, sample_response, wsgi_response):
+        """
+        Check that the sample response and wsgi response match.
+        """
+        self.assertEqual(wsgi_response.status_code, sample_response.status)
+        if 'body' in sample_response:
+            response_parsed = wsgi_response.data
+            self.assert_data_equal(response_parsed, sample_response.body)
+
+    def assert_matches_sample(self, path, label, response):
```

Comment:
Yes, that was one of the controversial things about this PR that was up for discussion (see the PR description). 

Should be relatively painless to fix - and I think you'll agree (assuming your other project is a django project) is less error prone. Before this change you could have posted anywhere you wanted, and just passed in a url that worked to the assertion.